indico/modules/events/registration/notifications.py

```diff
@@ -32,7 +35,15 @@ def notify_invitation(invitation, email_body, from_address):
 
 
 def _notify_registration(registration, template, to_managers=False):
-    template = get_template_module('events/registration/emails/{}'.format(template), registration=registration)
+    custom_message = ''
+    if registration.state == RegistrationState.pending:
+        custom_message = registration.registration_form.message_pending
+    elif registration.state == RegistrationState.unpaid:
+        custom_message = registration.registration_form.message_unpaid
+    elif registration.state == RegistrationState.complete:
+        custom_message = registration.registration_form.message_complete
+    template = get_template_module('events/registration/emails/{}'.format(template), registration=registration,
+                                   custom_message=Markup(urlize(escape(custom_message))))
```

Comment:
Just had a thought.. you could just `| urlize` the messages in the template and avoid adding all the extra code here. `urlize` already escapes the string before urlizing it so it should work fine (`|urlize|safe` otherwise)